sunbeam-python/sunbeam/commands/juju.py

```diff
@@ -324,11 +335,12 @@ def channel_update_needed(self, channel: str, new_channel: str) -> bool:
 class AddCloudJujuStep(BaseStep, JujuStepHelper):
     """Add cloud definition to juju client."""
 
-    def __init__(self, cloud: str, definition: dict):
+    def __init__(self, cloud: str, definition: dict, controller: str | None = None):
         super().__init__("Add Cloud", "Adding cloud to Juju client")
 
         self.cloud = cloud
         self.definition = definition
+        self.controller = controller
 
     def is_skip(self, status: Optional["Status"] = None) -> Result:
```

Comment:
This step is_skip is only checking for local clouds while it should check for clouds on the controller when it's an external controller

Reply:
Added the necessary checks in subsequent commit